src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/hmrcAddToAList.scala.html

```diff
@@ -0,0 +1,104 @@
+@*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.govukfrontend.views.Aliases.Hint
+@import uk.gov.hmrc.govukfrontend.views.html.components.{GovukButton, GovukRadios}
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.button.Button
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.content.Text
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.fieldset.{Fieldset, Legend}
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.radios.{RadioItem, Radios}
+@import uk.gov.hmrc.hmrcfrontend.views.helpers.MessageHelpers.{messagesForLanguage, pluralMessageKey}
+@import uk.gov.hmrc.hmrcfrontend.views.viewmodels.addtoalist.AddToAList
+
+@this(govukRadios: GovukRadios, govukButton: GovukButton)
+
+@(params: AddToAList)(implicit messages: Messages, messagesApi: MessagesApi)
```

Comment:
With that one my thought was that if we're already requiring an implicit messages api we might as well use the messages rather than just the language https://github.com/hmrc/play-frontend-hmrc/blob/2ba82c75f6284ece52a3a034996dbdd711058f0a/src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/hmrcCharacterCount.scala.html#L25 for this if I @inject messagesApi into the messagesForLanguages and make that a DI dependency then they wouldn't need to pass anything other than the implicit messages like the other components

Reply:
I'll update this to quickly to match what I was thinking as an example, but if that still isn't quite right after review then happy to simplify, won't be much work to try both first I think 👍